### PR TITLE
Restrict Proposer Lookup to the Current Epoch

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

- [x] Previously we were erroneously looking up proposer indexes for the next epoch, this PR disables that.
- [x] Reject zero-hash as a valid key. 
- [x] Add tests

**Which issues(s) does this PR fix?**

Part of fix for #7498 

**Other notes for review**

### CODE REVIEW

File: beacon-chain/core/helpers/committee.go [hunk 1]
```diff
@@ -370,6 +373,10 @@ func UpdateProposerIndicesInCache(state *stateTrie.BeaconState, epoch uint64) er
 	if err != nil {
 		return err
 	}
+	// Don't save zero-hash as key in cache.
+	if bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
+		return nil
```
Comment (rauljordan) [inline beacon-chain/core/helpers/committee.go:378]: is it ok to return nil or should we err?

File: beacon-chain/core/helpers/committee_test.go [hunk 2]
```diff
@@ -203,6 +203,36 @@ func TestCommitteeAssignments_CanRetrieve(t *testing.T) {
...
+
+	state, err := beaconstate.InitializeFromProto(&pb.BeaconState{
+		Validators:  validators,
+		Slot:        2 * params.BeaconConfig().SlotsPerEpoch, // epoch 2
+		RandaoMixes: make([][]byte, params.BeaconConfig().EpochsPerHistoricalVector),
+	})
+	require.NoError(t, err)
+	_, proposerIndxs, err := CommitteeAssignments(state, CurrentEpoch(state))
+	require.NoError(t, err)
+	require.NotEqual(t, 0, len(proposerIndxs), "wanted  non-zero proposer index set")
```
Comment (rauljordan) [inline beacon-chain/core/helpers/committee_test.go:229]:
```suggestion
	require.NotEqual(t, 0, len(proposerIndxs), "wanted non-zero proposer index set")
```

File: beacon-chain/core/helpers/validators.go [hunk 3]
```diff
@@ -186,12 +188,14 @@ func BeaconProposerIndex(state *stateTrie.BeaconState) (uint64, error) {
...
-		}
-		if proposerIndices != nil {
-			return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
+		if !bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
+			proposerIndices, err := proposerIndicesCache.ProposerIndices(bytesutil.ToBytes32(r))
+			if err != nil {
+				return 0, errors.Wrap(err, "could not interface with committee cache")
+			}
+			if proposerIndices != nil {
+				return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
```
Comment (rauljordan) [inline beacon-chain/core/helpers/validators.go:197]: Do we need a length check?

**Review by rauljordan — DISMISSED** [review]

File: beacon-chain/rpc/validator/assignments.go [hunk 4]
```diff
@@ -155,7 +155,6 @@ func (vs *Server) duties(ctx context.Context, req *ethpb.DutiesRequest) (*ethpb.
 
 			nextAssignment.ValidatorIndex = idx
 			nextAssignment.Status = assignmentStatus(s, idx)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator/assignments.go:157]: Can you add a comment here why nextAssignment doesn't have proposer slots?

Comment (nisdas) [reply beacon-chain/rpc/validator/assignments.go:158]: Done

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**